Flaky store tests: CreateDevice test helper generates colliding MACs under the accepted-mac unique index

## Summary

`TestPgStore/StatsStore` (and any store test that creates more than one accepted
device in the same namespace) is flaky. It fails intermittently with:

```
--- FAIL: TestPgStore/StatsStore/succeeds_with_specific_tenantID
    stats_tests.go:49:
        Error Trace: api/store/storetest/helpers.go:226
                     api/store/storetest/stats_tests.go:49
        Error: Received unexpected error:
               document duplicate
```

Seen on CI: https://github.com/shellhub-io/shellhub/actions/runs/28873799116/job/85643304102
A plain re-run of the same commit goes green, which is the tell-tale of a flake.

## Root cause

`CreateDevice` in `api/store/storetest/helpers.go` builds the default device MAC
from the wall clock:

```go
Identity: &models.DeviceIdentity{MAC: fmt.Sprintf(
    "%02x:%02x:%02x:%02x:%02x:%02x",
    time.Now().UnixNano()%256, time.Now().UnixNano()%256, ...)},
```

On a runner whose clock resolution is coarser than the six `time.Now()` calls,
all six octets collapse to the same value, so the whole MAC becomes
`VV:VV:VV:VV:VV:VV`. Two devices created back to back then collide on their MAC
with probability ~1/256 per pair.

That collision was harmless until migration `007_devices_accepted_mac_unique`
(#6600) added a unique index on `(namespace_id, mac) WHERE status = 'accepted'`.
Now a colliding MAC across two accepted devices in the same namespace turns into
a `document duplicate` on insert. `StatsStore/succeeds_with_specific_tenantID`
creates three accepted devices in `tenant1`, so it is the most frequent victim.

The default UID (`fmt.Sprintf("%064x", time.Now().UnixNano())`) has the same
weakness: two devices created within one clock tick collide on the primary key.

## Suggested fix

Give the test helper collision-free defaults instead of deriving them from the
clock. A package-level `atomic.Uint64` counter that feeds both the UID and the
MAC is enough:

```go
var deviceSeq atomic.Uint64

n := deviceSeq.Add(1)
uid := fmt.Sprintf("%064x", n)
mac := fmt.Sprintf("00:00:%02x:%02x:%02x:%02x", byte(n>>24), byte(n>>16), byte(n>>8), byte(n))
```

Callers that need a specific MAC already pass one through a `DeviceOption`, so
only the default changes. The other clock-derived defaults in this file
(namespace name, user email, session UID) are candidates for the same treatment
if they start colliding under their own unique constraints.

## Notes

This is a pre-existing issue in the test helper on `master`, exposed by the
#6600 unique index. It is not tied to any one feature branch; it can fail any PR
whose run happens to hit the collision.

--- a/server/api/store/storetest/helpers.go
+++ b/server/api/store/storetest/helpers.go
@@ -5,6 +5,7 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -194,6 +195,21 @@ func WithDeviceRemoteAddr(addr string) DeviceOption {
 	}
 }
 
+// deviceSeq backs the default device identifiers. A counter rather than the clock because
+// devices are unique on (namespace_id, mac) while accepted, and clock-derived values collide
+// often enough to fail the suite.
+var deviceSeq atomic.Uint64
+
+func nextDeviceUID() string {
+	return fmt.Sprintf("%064x", deviceSeq.Add(1))
+}
+
+func nextDeviceMAC() string {
+	n := deviceSeq.Add(1)
+
+	return fmt.Sprintf("00:00:%02x:%02x:%02x:%02x", (n>>24)&0xFF, (n>>16)&0xFF, (n>>8)&0xFF, n&0xFF)
+}
+
 // CreateDevice creates a device with default or customized values
 // Returns the generated device UID
 // If tenant is not provided via WithTenantID(), a default namespace will be created
@@ -202,15 +218,11 @@ func (s *Suite) CreateDevice(t *testing.T, opts ...DeviceOption) models.UID {
 	ctx := context.Background()
 	st := s.provider.Store()
 
-	// Generate unique UID (sha256-like format)
-	uid := fmt.Sprintf("%064x", time.Now().UnixNano())
-
-	// Default device
 	device := &models.Device{
-		UID:       uid,
+		UID:       nextDeviceUID(),
 		Name:      fmt.Sprintf("device_%d", time.Now().UnixNano()),
 		TenantID:  "", // Will be set below if not provided via options
-		Identity:  &models.DeviceIdentity{MAC: fmt.Sprintf("%02x:%02x:%02x:%02x:%02x:%02x", time.Now().UnixNano()%256, time.Now().UnixNano()%256, time.Now().UnixNano()%256, time.Now().UnixNano()%256, time.Now().UnixNano()%256, time.Now().UnixNano()%256)},
+		Identity:  &models.DeviceIdentity{MAC: nextDeviceMAC()},
 		Info:      &models.DeviceInfo{},
 		PublicKey: "-",
 		Status:    models.DeviceStatusAccepted,

--- a/server/api/store/storetest/helpers_test.go
+++ b/server/api/store/storetest/helpers_test.go
@@ -1,0 +1,40 @@
+package storetest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNextDeviceIdentifiersAreDistinct(t *testing.T) {
+	const calls = 1024
+
+	for _, tc := range []struct {
+		description string
+		next        func() string
+	}{
+		{
+			description: "UID",
+			next:        nextDeviceUID,
+		},
+		{
+			description: "MAC",
+			next:        nextDeviceMAC,
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			seen := make(map[string]int, calls)
+
+			for i := range calls {
+				got := tc.next()
+				if first, dup := seen[got]; dup {
+					t.Fatalf("call %d produced %q, already produced by call %d", i, got, first)
+				}
+
+				seen[got] = i
+			}
+
+			assert.Len(t, seen, calls)
+		})
+	}
+}

